server/src/test/omtLinkProvider.test.ts

```diff
@@ -0,0 +1,132 @@
+            actualDocumentLinks = linkProvider.provideDocumentLinks(textDocument);
+            //- called in the catch clause for reading tsconfig-invalid
+            expect(errorStub).to.be.calledOnce;
+            errorStub.restore();
+        })
+
+        // test uri links with and without shorthands
+        const cases: Case[] = [
+            { i: 0, line: 1, start: 1, end: 18, target: undefined, should: 'should make a link for a declared import', module: 'Declared' },
+            { i: 1, line: 3, start: 5, end: 20, target: resolve('testFixture/relative.omt'), should: 'should make a link for a relative path' },
+            { i: 2, line: 5, start: 4, end: 24, target: resolve('testFixture/one/shorthanded.omt'), should: 'should recognize a shorthand from the tsconfig in current project folder' },
+            { i: 3, line: 7, start: 5, end: 29, target: resolve('testFixture/one/quotedShorthand.omt'), should: 'should ignore quotes 1' },
+            { i: 4, line: 9, start: 5, end: 35, target: resolve('testFixture/one/doubleQuotedShorthand.omt'), should: 'should ignore quotes 2' },
+            { i: 5, line: 11, start: 4, end: 31, target: resolve('testFixture/one/@two/siblingShorthanded.omt'), should: 'should not ignore a shorthand from a sibling path' },
+            { i: 6, line: 13, start: 4, end: 27, target: resolve('testFixture/three/parentPathed.omt'), should: 'should recognize a shorthand from the tsconfig in a parent project' },
+        ];
+
+        // test all cases
+        cases.forEach((value: Case) => {
+            it(value.should, () => {
+                const { module } = value;
+                testLinkProvider(
+                    value.i,
+                    {
+                        range: toRange(value.line, value.start, value.line, value.end),
+                        target: value.target,
+                        data: module ? { declaredImport: { module } } : undefined,
+                    });
+            })
+        });
+
+        function testLinkProvider(index: number, expectedDocumentLink: DocumentLink) {
+            const actualDocumentLink = actualDocumentLinks[index];
+            if (expectedDocumentLink.target) {
+                expect(actualDocumentLink.target).to.equal(expectedDocumentLink.target, 'target path');
+                expect(actualDocumentLink.range).to.deep.equal(expectedDocumentLink.range, 'range');
+                expect(actualDocumentLink.data).to.be.undefined;
+            } else {
+                expect(actualDocumentLink.target).to.equal(undefined, 'target');
+                expect(actualDocumentLink.data).not.to.be.undefined;
+                expect(actualDocumentLink.data.declaredImport.module).to.eq(expectedDocumentLink.data.declaredImport.module);
+            }
+        }
+    });
+
+    describe('resolveLink', () => {
+        it('should resolve declared imports', () => {
+            const functionStub = stub(lookupStub, 'getModulePath').returns('modulePath');
```

Comment:
Put this in a `beforeEach`, because it is done in each of the three tests in this `describe`

Reply:
done